### DISCUSSION TITLE
Allow ungrouped navigation items to be placed between / below navigation groups

### DISCUSSION THREAD

**Discussion by binaryfire** [filamentphp/filament#10130] [Ideas]
Hey @danharrin and @zepfietje. Pinging you because I'm willing to work on a PR for this but want some feedback first.

Unless I'm missing something, there's currently no way of placing ungrouped items between or below groups? There are a lot of situations where a single resource or `NavigationItem` represents an entire feature on its own. Here's an example from WordPress. This isn't possible in Filament atm:
<img width="368" alt="image" src="https://github.com/filamentphp/filament/assets/5361908/c66dc749-d511-4dce-8a78-e4664bbb0525">

Currently the only way to order standalone (ungrouped) items relative to groups is to create a group with a single entry, which looks pretty weird. I'm thinking we could add a `->sort()` method to `NavigationGroup` and then order the parent items in the final nav array using their sort orders. That would mean an standalone item with a sort order of 2 would appear under a group with with a sort order of 1. What do you think?

**Comment by zepfietje** [comment]
Judging from your screenshot, this seems like the nested navigation items feature introduced in 3.1 instead of groups. 🙂

**Reply by binaryfire** [reply]
@zepfietje Isn't that just for sub-grouping inside a group?

Let's say I have this Navigation group with some sub-grouped stuff:

```
Users
    - Admins
          - Roles
          - Permissions
   - Members           
```

And I have these 2 ungrouped standalone resources:
```
Themes
Orders
```

How could I create this navigation order?

```
1. Orders

2. Users
    - Admins
          - Roles
          - Permissions
   - Members           

3. Themes
```

**Reply by binaryfire** [reply]
From what I can see, `navigationParentItem` has to be a group: https://github.com/filamentphp/filament/blob/8f0ca0ea2be61d15864d60d6d2616f1539e5b51f/packages/panels/src/Resources/Resource.php#L731 If I try nesting under a parent that doesn't have a group I get an exception.

So I don't think the nesting feature helps with what I'm talking about here.

**Reply by danharrin** [reply]
`navigationParentItem()` is the label of another item

**Reply by binaryfire** [reply]
Sorry that exception was being caused by something else. Ok that works. But using nesting to solve this means you can't have subgroups. I.e. this still isn't possible:

```
1. Orders (item)

2. Users (group with nesting)
    - Admins
          - Roles
          - Permissions
    - Members           

3. Themes (item)
```

Only this would be possible:

```
1. Orders (item)

2. Users (parent item with nested items)
    - Admins
    - Members           

3. Themes (item)
```

What I'm looking for is a way to order all top level nav items collectively regardless of their type (group or item).

**Reply by zepfietje** [reply]
This means 4 levels of nesting in navs, which we're not going to support (neither do any big platforms with a proper design system that we've studied). What you're outlining above is different from the screenshot you posted though.

**Reply by binaryfire** [reply]
I don't think I'm explaining myself very well. About to submit a PR which will make it clearer.  Have a look at that and see what you think.

**Comment by giacomomasseron** [comment]
Sorry to replay to a closed post, but I have the exact same problem and I do not know how to solve it.

What I need it is a menu like this:

```
1. Orders (item)

2. Users (parent item with nested items)
    - Admins
    - Members           

3. Themes (item)
```
Right now every ungrouped item is on top of the navigation menu

**Reply by zepfietje** [reply]
You should be able to achieve your desired navigation by registering items and groups manually: https://filamentphp.com/docs/3.x/panels/navigation#registering-custom-navigation-items.
